libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java

```diff
@@ -106,8 +101,8 @@ public MapboxNavigation(@NonNull Context context, @NonNull String accessToken) {
   public MapboxNavigation(@NonNull Context context, @NonNull String accessToken,
                           @NonNull MapboxNavigationOptions options) {
     this.accessToken = accessToken;
-    this.context = context;
     this.options = options;
+    initializeContext(context);
```

Comment:
Because `initializeContext` could potentially throw an exception it's preferable to check it in the first place, avoiding any allocations if it finally is not a valid context.